src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1588,4 +1595,214 @@ public static void SetTestHook(string property, bool value)
             }
         }
     }
+
+#if CORECLR && !UNIX
+    /// <summary>
+    /// Helper to start process using ShellExecuteEx. This is used only in PowerShell Core on Full Windows.
+    /// </summary>
+    internal class ShellExecuteHelper
+    {
+        /// <summary>
+        /// Start a process using ShellExecuteEx with default settings about WindowStyle and Verb.
+        /// </summary>
+        internal static Process Start(ProcessStartInfo startInfo)
+        {
+            return Start(startInfo, ProcessWindowStyle.Normal, string.Empty);
+        }
+
+        /// <summary>
+        /// Start a process using ShellExecuteEx
+        /// </summary>
+        /// <remarks>
+        /// Quoted from MSDN:
+        /// "Because ShellExecuteEx can delegate execution to Shell extensions (data sources, context menu handlers, verb implementations) 
+        ///  that are activated using Component Object Model (COM), COM should be initialized before ShellExecuteEx is called. Some Shell
+        ///  extensions require the COM single-threaded apartment (STA) type. In that case, COM should be initialized as shown here:
+        ///     CoInitializeEx(NULL, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE)
+        ///  There are instances where ShellExecuteEx does not use one of these types of Shell extension and those instances would not require
+        ///  COM to be initialized at all. Nonetheless, it is good practice to always initalize COM before using this function."
+        ///
+        /// TODO: In .NET Core, managed threads are all eagerly initialized with MTA mode, so to call 'ShellExecuteEx' from a STA thread, we
```

Comment:
> #2969 will be used to track the "invoke-on-STA-thread" work.

Yes, it's recorded in the description of this PR. This `Start` method will be touched when fixing that issue and this `TODO` can be removed then.